content/fast/adaptive-serving-based-on-network-quality/codelab-adaptive-serving-based-on-network-quality.md

```diff
@@ -0,0 +1,172 @@
+---
+page_type: glitch
+title: Adaptive serving based on network quality
+author: mihajlija
+description: |
+  In this codelab you’ll learn how to use the Network Information API to adapt
+  your content based on the network quality.
+web_updated_on: 2019-01-31
+web_published_on: 2019-01-31
+glitch: adaptive-serving-netinfo-starter
+---
+
+# Adaptive serving based on network quality
+
+In this codelab you’ll learn how to adapt your content based on the network quality. This page’s background video should load only when users are on a fast network. On slower networks an image should load instead.
+
+The [Network Information API](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation) enables you to access information about the user's connection quality. You will use its `effectiveType` property to decide when to serve a video and when to serve an image. `effectiveType` can be ‘slow-2g’, ‘2g’, ‘3g’, or ‘4g’.
+
+## Step 1: Check for browser support
+
+The background video is currently specified index.html and then lazy-loaded in `script.js`.
+
+To load it conditionally, first check if connection type detection is supported. In `script.js` add an if statement that tests whether the `navigator.connection` object exists and whether it has the effectiveType property.
+
+```if (navigator.connection && !!navigator.connection.effectiveType) {}```
+
+Wrap the existing video loading code in an else statement, so that video will still load in browsers that don’t support the Network Information API.
+
+```
```

Comment:
This first step feels a bit misleading to me. I'd suggest removing it all together, and instead have the 'Check connection type' section cover browser support and connection type. Audience will understand that placeholder code way better than the sort of wiggly presentation of checking browser support without any conditions.